File: Day09_PicturesAndMenu/app/src/main/java/com/example/day09_picturesandmenu/MainActivity.java
package com.example.day09_picturesandmenu;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button btnContext = findViewById(R.id.btnContext);
        btnContext.setOnCreateContextMenuListener(this);
    }

    public void clickToGallery(View view) {
        Intent intent = new Intent(this, GalleryActivity.class);
        startActivity(intent);
    }

    private boolean createMenu(Menu menu) {
        menu.add(0, 0, 0, "Item 1");
        menu.add(0, 1, 1, "Item 2");
        menu.add(0, 2, 2, "Item 3");
        return true;
    }

    private boolean menuChoice(MenuItem item) {
        switch (item.getItemId()) {
            case 0:
                Toast.makeText(this, "Item 1 clicked", Toast.LENGTH_LONG).show();
                return true;
            case 1:
                Toast.makeText(this, "Item 2 clicked", Toast.LENGTH_LONG).show();
                return true;
            case 2:
                Toast.makeText(this, "Item 3 clicked", Toast.LENGTH_LONG).show();
                return true;
        }
        return false;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return createMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        return menuChoice(item);
    }

    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        createMenu(menu);
    }

    @Override
    public boolean onContextItemSelected(MenuItem item) {
        return menuChoice(item);
    }
}

File: kotlin/Day09/app/src/main/java/com/example/day09/MainActivity.kt
package com.example.day09

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.*
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private val slides = listOf<Int>(
        R.drawable.pic1,
        R.drawable.pic2,
        R.drawable.pic3,
        R.drawable.pic4
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button1.setOnCreateContextMenuListener(this)
        button2.setOnCreateContextMenuListener { menu, _, _ ->
            menu.add(0, 0, 0, "Annonymous menu 1")
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menu!!.add(0, 0, 0, "Option 1")
        menu!!.add(0, 0, 0, "Option 2")
        menu!!.add(0, 0, 0, "Option 3")
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        Toast.makeText(this, item!!.title, Toast.LENGTH_LONG).show()
        return super.onOptionsItemSelected(item)
    }

    override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
        menu!!.add(0,0,0, "Context 1")
        menu!!.add(0,0,0, "Context 2")
        menu!!.add(0,0,0, "Context 3")
        super.onCreateContextMenu(menu, v, menuInfo)
    }

    override fun onContextItemSelected(item: MenuItem?): Boolean {
        Toast.makeText(this, item!!.title, Toast.LENGTH_LONG).show()
        return super.onContextItemSelected(item)
    }
}
